Backend/nix/services.nix

```diff
@@ -0,0 +1,122 @@
+{ inputs, self', ... }:
+{
+  config = {
+    perSystem = { inputs', self', pkgs, lib, ... }: {
+      process-compose."services" = { config, ... }:
+        let
+          startScript = masterDb: replicaDb: port: pkgs.writeShellApplication {
```

Comment:
Could you add a comment above describing what this function actually does? Is it only used in backup processes? 

Wait, hold on ... is pg backup something we can upstream to services-flake? 

i.e., instead of:

```nix
settings.processes.pgBaseBackupForLocationDb = {
  command = "${startScript "location-db" "location-db-replica" "5454"}/bin/start-pgbasebackup";
  depends_on."location-db".condition = "process_healthy";
};
```

wouldnt' it be better to be able to do:

```nix
settings.postgres-backup.pgBaseBackupForLocationDb = {
  master = "location-db";
  replica = "location-db-replica";
  # port and depends_on can be inferred
};
```

?

As a first step, for this PR, perhaps this backup service can be defined locally in a separate `.nix` file ... acting, also, as a demonstrate of how to define custom service-flake modules?

Reply:
To begin with we can have a module in nammayatri, similar to [passetto](https://github.com/nammayatri/passetto/blob/nixify/process-compose.nix) and a supporting doc in `services-flake`. This could be added as a service if we get a feature request, as I don’t see a general use case of needing a replica DB running for local development.

Reply:
[postgres-with-replica](https://github.com/nammayatri/nammayatri/blob/24943fc5d47376b55ec4096d52266a9e96f53016/Backend/nix/postgres-with-replica.nix) is now a separate module.

[nix/services.nix](https://github.com/nammayatri/nammayatri/blob/24943fc5d47376b55ec4096d52266a9e96f53016/Backend/nix/services.nix) looks way cleaner.